samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinatorFactory.java

```diff
@@ -33,6 +33,7 @@
 public class ZkJobCoordinatorFactory implements JobCoordinatorFactory {
 
   private static final Logger LOG = LoggerFactory.getLogger(ZkJobCoordinatorFactory.class);
+  private static final String JOB_COORDINATOR_ZK_PATH_SUFFIX = "/JobCoordinatorData";
```

Comment:
QQ: This may probably start another debate. But I am throwing it out there anyways. 

In the future when we have more than one job for the same `StreamApplication`, we will run into the same path collision issue. Is this ever going to be a valid use-case where we define each node  in the DAG to be a separate **job** in itself? If that is the case, we will need another hierarchy when defining the scope - as in:
* /application/RunnerData (one per StreamApplication)
* /application/<StageJob>/JobCoordinatorData (one per stage)

@nickpan47  can you please clarify if this is the direction we are heading for in the long term?